apps/passport-server/src/routing/routes/genericIssuanceRoutes.ts

```diff
@@ -567,6 +570,59 @@ export function initGenericIssuanceRoutes(
     }
   );
 
+  /**
+   * For a given pipeline, gets an anonymized mapping from hashed email to
+   * Pretix order code, which is used for one-click email verification. Hit by
+   * the zupass backend, and authenticated with the DEVCON_PODBOX_API_KEY env
+   * var. Used on the zupass backend to authenticate one-click login requests.
+   */
+  app.get(
+    "/generic-issuance/api/one-click-emails/:pipelineId/:apiKey",
+    async (req, res) => {
+      checkGenericIssuanceServiceStarted(genericIssuanceService);
+      const pipelineId = checkUrlParam(req, "pipelineId");
+      const apiKey = checkUrlParam(req, "apiKey");
+
+      if (
+        !process.env.DEVCON_PODBOX_API_KEY ||
+        apiKey !== process.env.DEVCON_PODBOX_API_KEY
+      ) {
+        throw new PCDHTTPError(401, "invalid api key");
+      }
+
+      const pipeline = (
+        await genericIssuanceService.getAllPipelineInstances()
+      ).find((p) => p.id === pipelineId && PretixPipeline.is(p)) as
+        | PretixPipeline
+        | undefined;
+
+      if (!pipeline) {
+        throw new PCDHTTPError(400, "pipeline not found");
+      }
+
+      const tickets = await pipeline.getAllTickets();
+
+      const result: OneClickEmailResponseValue = { values: {} };
+
+      for (let i = 0; i < tickets.atoms.length; i++) {
+        const ticket = tickets.atoms[i];
+        if (ticket.email) {
+          const hashedEmail = sha256(ticket.email);
+          const hashedOrderCode = sha256(ticket.orderCode);
+          result.values[hashedEmail] = hashedOrderCode;
+        }
+
+        // throttle hashing to avoid locking up the server
+        // during this operation
+        if (i % 25 === 0) {
+          await sleep(1);
+        }
+      }
```

Comment:
interesting -- is there another way to do this? seems a little fishy to me xD

Reply:
I think it's fine - takes about 1s total.